homeassistant/components/stream/hls.py

```diff
@@ -0,0 +1,136 @@
+"""
+Provide functionality to stream HLS.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/stream/hls
+"""
+import datetime
+
+from aiohttp import web
+
+from .core import StreamView, StreamOutput
+
+
+async def async_setup_hls(hass):
```

Comment:
```suggestion
@callback
def async_setup_hls(hass):
```
it is also weird return a CONST string. If I were you, I will define the url format in module level and just import that in caller.

Reply:
In my opinion (and it's just that), that would make it harder to add additional stream formats in the future.

Reply:
We define our own setup format here, it's not standardized. New formats can do their own thing too. I would just make it callback.